src/sensai/vector_model.py

```diff
@@ -334,6 +247,23 @@ def predict(self, x: pd.DataFrame) -> pd.DataFrame:
     def _predict(self, x: pd.DataFrame) -> pd.DataFrame:
         pass
 
+    def _underlyingModelRequiresFitting(self) -> bool:
+        """
+        Designed to be overridden for rule-based models.
+
+        :return: True iff the underlying model requires fitting
+        """
+        return True
+
+    def fitPreprocessors(self, X: pd.DataFrame, Y: pd.DataFrame = None):
+        # no need for fitGenerate if chain is empty
+        if self._featureGenerator is not None:
+            if len(self._inputTransformerChain) == 0:
+                self._featureGenerator.fit(X, Y)
+            else:
+                X = self._featureGenerator.fitGenerate(X, Y, self)
+        self._inputTransformerChain.fit(X)
+
     def fit(self, X: pd.DataFrame, Y: pd.DataFrame, fitPreprocessors=True, fitTargetTransformer=True):
```

Comment:
I have to be able to fit rule based models without passing Y. Would it be ok for you to make Y None by default? Otherwise the user has to remember to explicitly pass None. Docu for Y also has to be adjusted

Reply:
You can't do this in VectorModel because it would suggest an incorrect interface to users.
You could do it in rule-based subclasses though.

Reply:
Sounds good. Anyway fit requires some special docu in the rule based stuff, this gives the possibility to add it